functions/Backup-DbaDatabase.ps1

```diff
@@ -453,8 +453,17 @@ function Backup-DbaDatabase {
 
             if ($CompressBackup) {
                 if ($db.EncryptionEnabled) {
-                    Write-Message -Level Warning -Message "$dbName is enabled for encryption, will not compress"
-                    $backup.CompressionOption = 2
+                    $minVerForTDECompression = [version]'13.0.4446.0' #SQL Server 2016 CU 4
+                    $flagTDESQLVersion = $minVerForTDECompression -le $Server.version
+                    $flagTestBoundMaxTransferSize = Test-Bound 'MaxTransferSize'
+                    $flagCorrectMaxTransferSize = $flagTestBoundMaxTransferSize -and ($MaxTransferSize -gt 64kb)
+                    if ($flagTDESQLVersion -and $flagTestBoundMaxTransferSize -and $flagCorrectMaxTransferSize) {
+                        Write-Message -Level Verbose "$dbName is enabled for encryption but will compress"
```

Comment:
Thanks for including verbose messages - I like that a lot. Maybe add the "-Message" parameter name like in the other messages? And after all: The change looks good. Let's see what @potatoqualitee  says.